spec/models/user_spec.rb

```diff
@@ -645,4 +644,34 @@
       end
     end
   end
+
+  describe '#update_email_in_mailchimp' do
+    let(:email) { 'old@example.org'}
+    let(:user) { create(:user, email: email) }
+
+    context 'when email not changed' do
+      it 'does not update mailchimp' do
+        expect(Mailchimp::EmailUpdaterJob).not_to receive(:perform_later)
+        user.update!(name: 'New name')
+      end
+    end
+
+    context 'when email changed' do
+      context 'when user is not in mailchimp' do
+        it 'does not update mailchimp' do
+          expect(Mailchimp::EmailUpdaterJob).not_to receive(:perform_later)
+          user.update!(email: 'new@example.org')
+        end
+      end
+
+      context 'when user is in mailchimp' do
+        let(:user) { create(:user, email: email, mailchimp_status: :subscribed) }
+
+        it 'updates mailchimp' do
+          expect(Mailchimp::EmailUpdaterJob).to receive(:perform_later).with(user: user, original_email: email)
```

Comment:
I would have preferred the test to have run the job then check the user rather than mock

Reply:
Not sure if this is what you meant but have updated the spec. Am now:

- allowing the job to run, but stubbing the Mailchimp API instead
- have changed the job to set the `mailchimp_updated_at` date on the user, which is probably should have been anyway
- checking that the timestamp has been updated after the job runs